connectors/runner.py

```diff
@@ -94,6 +96,9 @@ def stop(self):
         self._sleeps.cancel()
         if self.connectors is not None:
             self.connectors.stop_waiting()
+        if self.job_cleanup_task is not None:
```

Comment:
this is a problematic pattern because you delete the object without waiting for the task it holds to finish.

you probably want to make shutdown() a coroutine, and wait on it, and inside wait for the task to finish once you stop it